web/src/SidebarIcon.tsx

```diff
@@ -0,0 +1,159 @@
+import React, { PureComponent } from "react"
+import { TriggerMode, RuntimeStatus } from "./types"
+import { Color } from "./constants"
+
+type SidebarIconProps = {
+  triggerMode: TriggerMode
+  status: RuntimeStatus
+  hasWarning: boolean
+  isBuilding: boolean
+}
+
+export enum IconType {
+  DotAuto = "dotAuto",
+  DotAutoPending = "dotAutoPending",
+  DotAutoBuilding = "dotAutoBuilding",
+  DotManual = "dotManual",
+  DotManualBuilding = "dotManualBuilding",
+}
+
+export default class SidebarIcon extends PureComponent<SidebarIconProps> {
+  render() {
+    let props = this.props
+    let fill = Color.green
+    if (props.status === RuntimeStatus.Pending) {
+      fill = Color.gray
+    } else if (props.status === RuntimeStatus.Error) {
+      fill = Color.red
+    } else if (props.hasWarning) {
+      fill = Color.yellow
+    }
+
+    if (props.triggerMode === TriggerMode.TriggerModeManual) {
+      return this.renderManual(fill)
+    }
+
+    return this.renderAuto(fill)
+  }
+
+  renderAuto(fill: Color) {
+    let props = this.props
+    if (props.isBuilding) {
+      return this.dotAutoBuilding(fill)
+    }
+
+    if (props.status === RuntimeStatus.Pending) {
+      return this.dotAutoPending(fill)
+    }
+
+    return this.dotAuto(fill)
+  }
+
+  renderManual(fill: Color) {
+    let props = this.props
+    if (props.isBuilding) {
+      return this.dotManualBuilding(fill)
+    }
+
+    return this.dotManualPending(fill)
+  }
+
+  dotAuto(fill: Color) {
+    return (
+      <svg
```

Comment:
i'm surprised you inlined the svg? it might be easier to import the svg from a separate file, then set these properties on the component. makes it easier to view them independently

i can show you how to do this if it's unclear

Reply:
I can take a crack at this. We sort of had a roundabout journey where we were looking at inlining SVG in a CSS `content` declaration, so we could design the icons purely in CSS. But gave up on that, but kept the inline SVG habit. I agree that we can import them as we do with other SVGs elsewhere.